src/frontend/sharedComponents/CustomHeaderLeft.tsx

```diff
@@ -86,27 +88,39 @@ const HeaderBackIcon = ({ tintColor }: { tintColor: string }) => (
  * Whether the observation has been edited is checked by deep-equal between the
  * draft and the original observation
  */
-const CustomHeaderLeft = ({ onPress: originalOnPress, ...props }: any) => {
+
+interface CustomHeaderLeftProps {
+  onPress?: () => void;
+  tintColor?: string;
+  headerBackButtonProps: HeaderBackButtonProps;
+}
+
+const CustomHeaderLeft = ({
+  onPress,
+  tintColor,
+  headerBackButtonProps,
+}: CustomHeaderLeftProps) => {
```

Comment:
I can't actually leave a comment on it, but it looks like there is a new way to prevent navigation (e.g. to show a "are you sure" warning) in RN6: https://reactnavigation.org/docs/preventing-going-back - this could be an alternative to our current implementation of `useBackHandler`, but I think it's ok to leave this to a follow-up PR.